src/fhirtypes/StructureDefinition.ts

```diff
@@ -347,6 +356,12 @@ export class StructureDefinition {
         pathPart.brackets?.forEach(p => (currentPath += `[${p}]`));
       }
       currentElement = this.findElementByPath(currentPath, resolve);
+
+      // If the element has a base.max that is great than 1, but the element has been constrained, still set properties in an array
+      const valueShouldBeWithinArray =
+        currentElement?.base?.max !== '0' &&
+        currentElement?.base?.max !== '1' &&
+        (currentElement?.max === '0' || currentElement?.max === '1');
```

Comment:
I'm confused about this last line.  So if the `currentElement.base.max` is `*` _and_ the `currentElement.max` is also `*`, we're setting `valueShouldBeWithinArray` to `false`?  Why?

The logic above also effectively says that if there is no `currentElement.base` defined, but `currentElement.max` is `*`, then we also say `valueShouldBeWithinArray` is `false`.  I don't think that's right.  If there is no `base`, then the `currentElement.max` should be used to determine if it's an array (e.g., if `currentElement.max` is not `0` or `1`).  Actually the opposite of how it works above if `currentElement.base` is not defined.

Reply:
I see it gets handled ok down below, but since this variable is called `valueShouldBeWithinArray`, it's a bit confusing that it's not always correct.

Reply:
I think maybe to clarify what this variable is actually doing, I will rename the variable to be more accurate. I'm thinking something along the lines of `baseMaxGreaterThanMax` or `constrainedElementBaseIsArray`. Neither of those exactly capture what the variable is checking, but I'm struggling to come up with a short name that fully captures the logic. @ngfreiter or @mint-thompson any ideas?

Reply:
Maybe `nonArrayElementBasedOnArray`. A mouthful but it makes sense I think.

Reply:
Oh yes! I like that!